File: bookvillage/src/main/java/com/mi/bookvillage/customer/model/service/CustomerService.java
package com.mi.bookvillage.customer.model.service;

import com.mi.bookvillage.customer.model.dao.CustomerDAO;
import com.mi.bookvillage.customer.model.vo.CustomerVO;
import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class CustomerService {

    private CustomerDAO customerDAO;
    private BCryptPasswordEncoder encoder;

    public CustomerService(CustomerDAO customerDAO , BCryptPasswordEncoder encoder){
        this.customerDAO = customerDAO;
        this.encoder = encoder;
    }

    public List<CustomerVO> getCustomerList(){
        return customerDAO.getCustomerList();
    }


    public CustomerVO LoginCustomer(CustomerVO customerVO){
        CustomerVO authCustomer = customerDAO.loginCustomer(customerVO);
        if(authCustomer == null || encoder.matches( customerVO.getUserPw(), authCustomer.getUserPw())){
            throw new RuntimeException("password can not be null or unmatched");
        }
        return authCustomer;
    }


    public void addCustomer(CustomerVO customerVO){
        String encodedPassword = encoder.encode(customerVO.getUserPw());
        customerVO.setUserPw(encodedPassword);
        customerDAO.addCustomer(customerVO);
    }


    public CustomerVO getCustomerDetail(int userSeq){
        CustomerVO customerVO = customerDAO.getCustomerBySeq(userSeq);
        return customerVO;
    }

    public void updateCustomer(CustomerVO customerVO){
        customerDAO.updateCustomer(customerVO);
    }

    public void deleteCustomer(int userSeq){
        customerDAO.deleteCustomer(userSeq);
    }

}

File: bookvillage/front-side/admin-front/src/service/customerService.js
import $axiosInst from '@/axios/AxiosInst'
import {adminResponseApi} from "@/util/AdminResponseApi";

export class customerService {

    constructor(host) {
        this.host = host;
    }

    getCustomerList(){
        return $axiosInst
            .get(`${this.host}/customer/list`)
            .then( response => {
                return response.data.data;
            }).catch( error => {
                throw adminResponseApi(error);
            })
    }


    addCustomer(customerObj){
        return $axiosInst
            .post(`${this.host}/customer/add` , customerObj)
            .then( response => {
                return response.data.data;
            }).catch( error => {
                throw adminResponseApi(error);
            })
    }

    getCustomerDetail(userSeq) {
        return $axiosInst
            .get(`${this.host}/customer/detail/` + userSeq)
            .then(response => {
                return response.data.data;
            }).catch(error => {
                throw adminResponseApi(error);
            })
    }

    updateCustomer(customerObj) {
        const userSeq = customerObj.userSeq
        return $axiosInst
            .put(`${this.host}/customer/update/` + userSeq , customerObj)
            .then( () => {
                return;
            }).catch( error => {
                throw adminResponseApi(error);
            })
    }

    deleteCustomer(userSeq){
        return $axiosInst
            .delete(`${this.host}/customer/delete/` + userSeq)
            .then( () => {
                return;
            }).catch( error => {
                throw adminResponseApi(error);
            })
    }



}